protocol/contracts/leaser/src/contract.rs

```diff
@@ -45,10 +48,9 @@ pub fn instantiate(
 
     ContractOwnerAccess::new(deps.storage.deref_mut()).grant_to(&info.sender)?;
 
-    let lease_code = msg.lease_code_id;
-    Config::new(msg).store(deps.storage)?;
-
-    leaser::update_lpp(deps.storage, lease_code.into(), Batch::default())
+    Config::try_new(msg, &deps.querier)
+        .and_then(|config| config.store(deps.storage).map(|()| config))
+        .and_then(|config| leaser::update_lpp(deps.storage, config.lease_code, Batch::default()))
```

Comment:
[Suggestion] The two `Result::and_then` calls could be nested to avoid the need for the `Result::map` call.

Reply:
prefer to keep the functional expressions as shallow as possible, up to some complexity extent for sure :-)